include/aspect/mesh_deformation/interface.h

```diff
@@ -384,32 +410,44 @@ namespace aspect
         AffineConstraints<double> mesh_vertex_constraints;
 
         /**
-         * A set of boundary indicators that denote those boundaries that are
-         * allowed to move their mesh tangential to the boundary. All
-         * boundaries that have tangential material velocity boundary
-         * conditions are in this set by default, but it can be extended by
-         * open boundaries, boundaries with traction boundary conditions, or
-         * boundaries with prescribed material velocities if requested in
-         * the parameter file.
+         * A map of boundary ids to mesh deformation objects that have been requested
+         * in the parameter file.
          */
-        std::set<types::boundary_id> tangential_mesh_boundary_indicators;
+        std::map<types::boundary_id,std::vector<std::unique_ptr<Interface<dim> > > > mesh_deformation_objects;
 
         /**
          * Map from boundary id to a vector of names representing
          * mesh deformation objects.
          */
-        std::map<types::boundary_id, std::vector<std::string> > mesh_deformation_boundary_indicators_map;
+        std::map<types::boundary_id, std::vector<std::string> > mesh_deformation_boundary_object_names;
```

Comment:
Why is this not called mesh_deformation_object_names as the variable above?

Reply:
Copy-paste error probably. fixed now.